internal/testing/blocklist_test.go

```diff
@@ -18,53 +17,31 @@ func TestBlocklist(t *testing.T) {
 	ctx := context.Background()
 
 	// create a new test cluster
-	cluster, err := newTestCluster(t.TempDir(), newTestLogger())
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer func() {
-		if err := cluster.Shutdown(ctx); err != nil {
-			t.Fatal(err)
-		}
-	}()
+	cluster := newTestCluster(t, testClusterOptions{
+		hosts: 3,
+	})
+	defer cluster.Shutdown()
 	b := cluster.Bus
+	tt := cluster.tt
 
-	// add hosts
-	_, err = cluster.AddHostsBlocking(3)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	// wait until we have 3 contracts in the set
-	var contracts []api.ContractMetadata
-	if err := Retry(5, time.Second, func() (err error) {
-		contracts, err = b.ContractSetContracts(ctx, testAutopilotConfig.Contracts.Set)
-		if err != nil {
-			t.Fatal(err)
-		} else if len(contracts) != 3 {
-			err = fmt.Errorf("unexpected number of contracts, %v != 3", len(contracts))
-			return
-		}
-		return
-	}); err != nil {
-		t.Fatal(err)
+	// fetch contracts
+	contracts, err := b.ContractSetContracts(ctx, testAutopilotConfig.Contracts.Set)
+	tt.OK(err)
+	if len(contracts) != 3 {
```

Comment:
we'd have to resort to some seriously ugly generics, but having `tt.AssertLen` would be pretty epic